pandas/core/base.py

```diff
@@ -532,7 +533,15 @@ def to_numpy(
             )
 
         if na_value is not lib.no_default:
-            values = self._values.copy()
+            values = self._values
+            if not can_hold_element(values, na_value):
```

Comment:
can we avoid can_hold_element?  its inaccurate for some EA dtypes

Reply:
We don’t get here with ea dtype, there is another path a few lines earlier

Reply:
great thanks